src/foremast/utils/dns.py

```diff
@@ -148,6 +149,13 @@ def update_failover_dns_record(env, zone_id, **kwargs):
     zone_name = hosted_zone_info['HostedZone']['Name'].rstrip('.')
     dns_name = kwargs.get('dns_name')
 
+    #Check that the primary record exists
+    failover_state = kwargs.get('failover_state')
+    if failover_state.lower() != 'primary':
+        primary_exists = primary_record_exists(env, zone_id, dns_name)
+        if not primary_exists:
+            raise PrimaryDNSRecordNotFound("Primary Failover DNS record not found: {}".format(dns_name))
```

Comment:
Do you think its worth creating the primary record at this point since one does not exist?